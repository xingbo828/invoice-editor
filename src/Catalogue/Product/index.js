import { compose, withHandlers } from 'recompose'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { updateProduct } from '../../cartAction'
import { getProductQuantity } from '../../cartReducer'

import Product from './Product'

const mapDispatchToProps = dispatch => bindActionCreators({
  updateProduct
}, dispatch)

const mapStateToProps = (state, ownProps) => ({
  quantity: getProductQuantity(ownProps.productDetail.id)(state)
});

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),
  withHandlers({
    handleChange: props => (v) => {
      props.updateProduct({
        product: props.productDetail,
        quantity: v
      })
    }
  })
)

export default enhance(Product)

