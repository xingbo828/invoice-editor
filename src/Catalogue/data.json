[
  {
    "id": "sofa",
    "title": "Sofa",
    "price": 325,
    "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg"
  },
  {
    "id": "dishwasher",
    "title": "Dishwasher",
    "price": 799,
    "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG"
  },
  {
    "id": "washing_machine",
    "title": "Washing Machine",
    "price": 589,
    "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg"
  },
  {
    "id": "refrigerator",
    "title": "Refrigerator",
    "price": 650,
    "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG"
  },
  {
    "id": "tv",
    "title": "Television",
    "price": 1200,
    "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true"
  },
  {
    "id": "deep_freezere",
    "title": "Deep Freezer",
    "price": 425,
    "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg"
  }
]