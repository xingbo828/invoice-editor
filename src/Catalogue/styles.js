import styled from 'styled-components'

export const ProductsContainer = styled.ul`
  margin: ${props=>props.theme.spaces.wide} 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  ${props=>props.theme.media.greaterThan('sm')`
    justify-content: space-between;
    flex-direction: row;
    flex-wrap: wrap;
  `}
`