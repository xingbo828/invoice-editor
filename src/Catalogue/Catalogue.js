import React from 'react';

import Product from './Product';
import data from './data.json';

import { ProductsContainer } from './styles';

import Button from '../Shared/Button'
import Heading from '../Shared/Heading'
import Container from '../Shared/Container'

const Catalogue = ({ handleCheckout }) => {
  const renderProducts = products =>
    products.map(product => (
      <Product key={product.id} productDetail={product} unit="Quantity" />
    ));
  return (
    <Container>
      <Heading wrapperTag="h1" centered size="md">Product catalogue</Heading>
      <ProductsContainer>
        {renderProducts(data)}
      </ProductsContainer>
      <Button onClick={handleCheckout} style={{float: 'right'}}>Checkout</Button>
    </Container>
  );
};

export default Catalogue;
