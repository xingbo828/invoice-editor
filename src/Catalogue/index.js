import { compose, withHandlers } from 'recompose'
import { withRouter } from "react-router";

import Catalogue from './Catalogue'

const enhance = compose(
  withRouter,
  withHandlers({
    handleCheckout: props => e => {
      e.preventDefault();
      props.history.push({
        pathname: '/invoice'
      })
    }
  })
)

export default enhance(Catalogue)