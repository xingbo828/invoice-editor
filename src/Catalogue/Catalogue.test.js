import React from 'react';
import { shallow } from 'enzyme';
import Catalogue from './Catalogue';

it('should match snapshot', () => {
  const wrapper = shallow(<Catalogue handleCheckout={() => {}}/>);
  expect(wrapper).toMatchSnapshot()
});