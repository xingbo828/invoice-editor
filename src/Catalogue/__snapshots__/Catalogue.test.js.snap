// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Catalogue
    handleCheckout={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          centered={true}
          size="md"
          wrapperTag="h1"
        >
          Product catalogue
        </ForwardRef>,
        <ForwardRef>
          <Connect(withHandlers(Product))
            productDetail={
              Object {
                "id": "sofa",
                "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg",
                "price": 325,
                "title": "Sofa",
              }
            }
            unit="Quantity"
          />
          <Connect(withHandlers(Product))
            productDetail={
              Object {
                "id": "dishwasher",
                "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG",
                "price": 799,
                "title": "Dishwasher",
              }
            }
            unit="Quantity"
          />
          <Connect(withHandlers(Product))
            productDetail={
              Object {
                "id": "washing_machine",
                "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg",
                "price": 589,
                "title": "Washing Machine",
              }
            }
            unit="Quantity"
          />
          <Connect(withHandlers(Product))
            productDetail={
              Object {
                "id": "refrigerator",
                "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG",
                "price": 650,
                "title": "Refrigerator",
              }
            }
            unit="Quantity"
          />
          <Connect(withHandlers(Product))
            productDetail={
              Object {
                "id": "tv",
                "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true",
                "price": 1200,
                "title": "Television",
              }
            }
            unit="Quantity"
          />
          <Connect(withHandlers(Product))
            productDetail={
              Object {
                "id": "deep_freezere",
                "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg",
                "price": 425,
                "title": "Deep Freezer",
              }
            }
            unit="Quantity"
          />
        </ForwardRef>,
        <Button
          onClick={[Function]}
          squared={false}
          style={
            Object {
              "float": "right",
            }
          }
        >
          Checkout
        </Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centered": true,
          "children": "Product catalogue",
          "size": "md",
          "wrapperTag": "h1",
        },
        "ref": null,
        "rendered": "Product catalogue",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "WithTheme(MarketingHeading)",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "sofa",
                  "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg",
                  "price": 325,
                  "title": "Sofa",
                }
              }
              unit="Quantity"
            />,
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "dishwasher",
                  "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG",
                  "price": 799,
                  "title": "Dishwasher",
                }
              }
              unit="Quantity"
            />,
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "washing_machine",
                  "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg",
                  "price": 589,
                  "title": "Washing Machine",
                }
              }
              unit="Quantity"
            />,
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "refrigerator",
                  "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG",
                  "price": 650,
                  "title": "Refrigerator",
                }
              }
              unit="Quantity"
            />,
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "tv",
                  "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true",
                  "price": 1200,
                  "title": "Television",
                }
              }
              unit="Quantity"
            />,
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "deep_freezere",
                  "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg",
                  "price": 425,
                  "title": "Deep Freezer",
                }
              }
              unit="Quantity"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "sofa",
            "nodeType": "class",
            "props": Object {
              "productDetail": Object {
                "id": "sofa",
                "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg",
                "price": 325,
                "title": "Sofa",
              },
              "unit": "Quantity",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "dishwasher",
            "nodeType": "class",
            "props": Object {
              "productDetail": Object {
                "id": "dishwasher",
                "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG",
                "price": 799,
                "title": "Dishwasher",
              },
              "unit": "Quantity",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "washing_machine",
            "nodeType": "class",
            "props": Object {
              "productDetail": Object {
                "id": "washing_machine",
                "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg",
                "price": 589,
                "title": "Washing Machine",
              },
              "unit": "Quantity",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "refrigerator",
            "nodeType": "class",
            "props": Object {
              "productDetail": Object {
                "id": "refrigerator",
                "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG",
                "price": 650,
                "title": "Refrigerator",
              },
              "unit": "Quantity",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "tv",
            "nodeType": "class",
            "props": Object {
              "productDetail": Object {
                "id": "tv",
                "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true",
                "price": 1200,
                "title": "Television",
              },
              "unit": "Quantity",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "deep_freezere",
            "nodeType": "class",
            "props": Object {
              "productDetail": Object {
                "id": "deep_freezere",
                "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg",
                "price": 425,
                "title": "Deep Freezer",
              },
              "unit": "Quantity",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-VigVT",
            "isStatic": false,
            "rules": Array [
              "
  margin: ",
              [Function],
              " 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.ul",
          "render": [Function],
          "styledComponentId": "sc-VigVT",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Checkout",
          "onClick": [Function],
          "squared": false,
          "style": Object {
            "float": "right",
          },
        },
        "ref": null,
        "rendered": "Checkout",
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-fjdhpX",
        "isStatic": false,
        "rules": Array [
          "
  padding: ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-fjdhpX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            centered={true}
            size="md"
            wrapperTag="h1"
          >
            Product catalogue
          </ForwardRef>,
          <ForwardRef>
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "sofa",
                  "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg",
                  "price": 325,
                  "title": "Sofa",
                }
              }
              unit="Quantity"
            />
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "dishwasher",
                  "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG",
                  "price": 799,
                  "title": "Dishwasher",
                }
              }
              unit="Quantity"
            />
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "washing_machine",
                  "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg",
                  "price": 589,
                  "title": "Washing Machine",
                }
              }
              unit="Quantity"
            />
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "refrigerator",
                  "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG",
                  "price": 650,
                  "title": "Refrigerator",
                }
              }
              unit="Quantity"
            />
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "tv",
                  "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true",
                  "price": 1200,
                  "title": "Television",
                }
              }
              unit="Quantity"
            />
            <Connect(withHandlers(Product))
              productDetail={
                Object {
                  "id": "deep_freezere",
                  "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg",
                  "price": 425,
                  "title": "Deep Freezer",
                }
              }
              unit="Quantity"
            />
          </ForwardRef>,
          <Button
            onClick={[Function]}
            squared={false}
            style={
              Object {
                "float": "right",
              }
            }
          >
            Checkout
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "centered": true,
            "children": "Product catalogue",
            "size": "md",
            "wrapperTag": "h1",
          },
          "ref": null,
          "rendered": "Product catalogue",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "WithTheme(MarketingHeading)",
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Connect(withHandlers(Product))
                productDetail={
                  Object {
                    "id": "sofa",
                    "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg",
                    "price": 325,
                    "title": "Sofa",
                  }
                }
                unit="Quantity"
              />,
              <Connect(withHandlers(Product))
                productDetail={
                  Object {
                    "id": "dishwasher",
                    "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG",
                    "price": 799,
                    "title": "Dishwasher",
                  }
                }
                unit="Quantity"
              />,
              <Connect(withHandlers(Product))
                productDetail={
                  Object {
                    "id": "washing_machine",
                    "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg",
                    "price": 589,
                    "title": "Washing Machine",
                  }
                }
                unit="Quantity"
              />,
              <Connect(withHandlers(Product))
                productDetail={
                  Object {
                    "id": "refrigerator",
                    "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG",
                    "price": 650,
                    "title": "Refrigerator",
                  }
                }
                unit="Quantity"
              />,
              <Connect(withHandlers(Product))
                productDetail={
                  Object {
                    "id": "tv",
                    "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true",
                    "price": 1200,
                    "title": "Television",
                  }
                }
                unit="Quantity"
              />,
              <Connect(withHandlers(Product))
                productDetail={
                  Object {
                    "id": "deep_freezere",
                    "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg",
                    "price": 425,
                    "title": "Deep Freezer",
                  }
                }
                unit="Quantity"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "sofa",
              "nodeType": "class",
              "props": Object {
                "productDetail": Object {
                  "id": "sofa",
                  "image": "https://hips.hearstapps.com/edc.h-cdn.co/assets/16/38/blue-sofa-joybird_1.jpg",
                  "price": 325,
                  "title": "Sofa",
                },
                "unit": "Quantity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "dishwasher",
              "nodeType": "class",
              "props": Object {
                "productDetail": Object {
                  "id": "dishwasher",
                  "image": "https://www.ikea.com/ca/en/images/products/lagan-built-in-dishwasher-white__0539685_PE653659_S4.JPG",
                  "price": 799,
                  "title": "Dishwasher",
                },
                "unit": "Quantity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "washing_machine",
              "nodeType": "class",
              "props": Object {
                "productDetail": Object {
                  "id": "washing_machine",
                  "image": "http://www.washing-machine-wizard.com/images/frigidaire-faqe7077kn-affinity-frontloader-electric-dryer-washing-machine-70-cubic-feet-21486932.jpg",
                  "price": 589,
                  "title": "Washing Machine",
                },
                "unit": "Quantity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "refrigerator",
              "nodeType": "class",
              "props": Object {
                "productDetail": Object {
                  "id": "refrigerator",
                  "image": "https://www.ikea.com/ca/en/images/products/lagan-top-freezer-white__0530524_PE646734_S4.JPG",
                  "price": 650,
                  "title": "Refrigerator",
                },
                "unit": "Quantity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "tv",
              "nodeType": "class",
              "props": Object {
                "productDetail": Object {
                  "id": "tv",
                  "image": "https://thegoodguys.sirv.com/products/50047364/50047364_508160.PNG?scale.height=505&scale.width=773&canvas.height=505&canvas.width=773&canvas.opacity=0&format=png&png.optimize=true",
                  "price": 1200,
                  "title": "Television",
                },
                "unit": "Quantity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "deep_freezere",
              "nodeType": "class",
              "props": Object {
                "productDetail": Object {
                  "id": "deep_freezere",
                  "image": "https://multimedia.bbycastatic.ca/multimedia/products/500x500/123/12353/12353893.jpg",
                  "price": 425,
                  "title": "Deep Freezer",
                },
                "unit": "Quantity",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-VigVT",
              "isStatic": false,
              "rules": Array [
                "
  margin: ",
                [Function],
                " 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.ul",
            "render": [Function],
            "styledComponentId": "sc-VigVT",
            "target": "ul",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Checkout",
            "onClick": [Function],
            "squared": false,
            "style": Object {
              "float": "right",
            },
          },
          "ref": null,
          "rendered": "Checkout",
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-fjdhpX",
          "isStatic": false,
          "rules": Array [
            "
  padding: ",
            [Function],
            ";
  margin: ",
            [Function],
            ";
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-fjdhpX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
